pkg/chunk/chunk.go

```diff
@@ -308,37 +308,8 @@ func chunksToIterators(chunks []Chunk) ([]local.SeriesIterator, error) {
 	return iterators, nil
 }
 
-// ChunksToMatrix converts a slice of chunks into a model.Matrix.
-func ChunksToMatrix(chunks []Chunk) (model.Matrix, error) {
-	// Group chunks by series, sort and dedupe samples.
-	sampleStreams := map[model.Fingerprint]*model.SampleStream{}
-	for _, c := range chunks {
-		fp := c.Metric.Fingerprint()
-		ss, ok := sampleStreams[fp]
-		if !ok {
-			ss = &model.SampleStream{
-				Metric: c.Metric,
-			}
-			sampleStreams[fp] = ss
-		}
-
-		samples, err := c.samples()
-		if err != nil {
-			return nil, err
-		}
-
-		ss.Values = util.MergeSampleSets(ss.Values, samples)
-	}
-
-	matrix := make(model.Matrix, 0, len(sampleStreams))
-	for _, ss := range sampleStreams {
-		matrix = append(matrix, ss)
-	}
-
-	return matrix, nil
-}
-
-func (c *Chunk) samples() ([]model.SamplePair, error) {
+// Samples returns all SamplePairs for the chunk.
+func (c *Chunk) Samples() ([]model.SamplePair, error) {
```

Comment:
This was made public because `chunksToMatrix` was moved to the `ingester` package which used `.samples()`. However I can move `ChunksToMatrix` back to the chunk package and expose it there as suggested.